src/user/user.service.ts

```diff
@@ -84,33 +130,27 @@ export class UserService {
           ...data,
           User: {
             connect: {
-              id: createUserDto.userId,
+              id: user.id,
             },
           },
         },
       });
     } catch (error) {
-      throw new HttpException('Failed to create user.', HttpStatus.INTERNAL_SERVER_ERROR);
+      if (error instanceof NotFoundException) throw error;
+      throw new GlobalException(ErrorsTypes.USER_PROFILE_FAILED_TO_ADD);
     }
   }
 
-  async updateUserProfile(id: string, updateUserDto: UpdateUserDto) {
+  async updateUserProfile(id: string, updateUserDto: UpdateUserDto, user: User) {
     try {
-      if (!updateUserDto.userId) {
-        throw new NotFoundException(`User with ID ${updateUserDto.userId} not found`);
+      if (!user) {
+        throw new NotFoundException("User doesn't exist");
```

Comment:
looks like forbidden exception. Is there possible a situation when we don't have user?

Reply:
we get it for directive CurrentUser
  updateProfile(
    @Param('id') id: string,
    @Body() updateUserDto: UpdateUserDto,
    @CurrentUser() user: User
  ) {
    return this.userService.updateUserProfile(id, updateUserDto, user);
so, it should be handled by validator or by guard if we have rule in buiseness logic that only registered users can update others profile. As I understand, only owner and admin can have such permissions.